### ISSUE TITLE
Can not recommend expect result of matrix

### ISSUE THREAD

**Issue by YuzhouPeng** [guoguibing/librec#237]
I have train and test set of |userid|itemid|exist/non-exist|matrix and I want to predict the value of item 7
This is my trainset
![capture](https://user-images.githubusercontent.com/13601004/35911178-fc63d0b6-0bf0-11e8-974e-dfc62fd432b0.PNG)
This is my test set
![capture1](https://user-images.githubusercontent.com/13601004/35911200-0dc8ea80-0bf1-11e8-8790-204fcf1285d7.PNG)
I configured the librec.properties file and import my train set and test set separately.
![capture2](https://user-images.githubusercontent.com/13601004/35911312-631b3448-0bf1-11e8-9bcd-341d52eeeb98.PNG)

However, I only got 0 rmse using biasedmf and svdpp recommender, and there is no prediction result
![capture3](https://user-images.githubusercontent.com/13601004/35911416-a70362c0-0bf1-11e8-9678-ebeae2a17905.PNG)

Can you tell me how to config file to get the result? Thanks a lot.

**Comment by SunYatong** [comment]
Hi, @YuzhouPeng 
- First of all, you shouldn't incorporate "non-exist" terms in your dataset explicitly.
- Secondly, BiasedMF is not suitable for your task, I recommend to try the algorithms without bias terms, like PMF.
- Thirdly, its still wired to conduct rating predictions with implicit feedback, maybe you should treat it as a ranking task by restricting the predictions of PMF into (0,1) with sigmoid function and training it with a cross entropy loss.

**Comment by YuzhouPeng** [comment]
Thanks for your answer.
However, I tried to remove non-exist term in my dataset and implemented pmf recommender, I still got 0 and there is no prediction result.
Train set
![capture1](https://user-images.githubusercontent.com/13601004/35916045-f7ede412-0c00-11e8-901a-813f85966971.PNG)

Test set
![capture](https://user-images.githubusercontent.com/13601004/35916050-fb84c2ee-0c00-11e8-8aed-989ac6055e19.PNG)
Result
![capture2](https://user-images.githubusercontent.com/13601004/35916169-6a370e0e-0c01-11e8-943a-e4d4f65ba297.PNG)
![capture3](https://user-images.githubusercontent.com/13601004/35916170-6bd53786-0c01-11e8-829f-d15ff4b44f02.PNG)

**Comment by SunYatong** [comment]
Hi, I remember that we have fixed an issue #228 about running rating prediction models with implicit feedback datasets. But we didn't modify the .jar file. I just updated it #239 and it will takes about 40 min to pass the tests.

**Comment by YuzhouPeng** [comment]
Thanks for your answer.
I ran pmf and got the result as following
![capture](https://user-images.githubusercontent.com/13601004/35970179-52f28e88-0cc1-11e8-8d62-528e53307674.PNG)

However,  I don't understand 'restricting the predictions of PMF into (0,1) with sigmoid function and training it with a cross entropy loss', I try to search on google but I cannot get any related result.
Could you give me some instruction? 
Thanks a million!

**Comment by SunYatong** [comment]
These outputs are already reasonable, there is no need to modify the source code.

You can google the correlation between linear regression and logistic regression, which is just the same as that between raw PMF and PMF with sigmoid+cross entropy.
